### DISCUSSION TITLE
Map Visualizations are empty

### DISCUSSION THREAD

**Discussion by RagnarSTS** [Security-Onion-Solutions/securityonion#2972] [Unsupported Versions] [answered]
On 2.3.x and all the map visualizations are empty. 

I am afraid searching the documentation yields answers that only make more questions.  Everything ~seems~ ok on the data front and indexes. 

Apologies if this is figured out and documented somewhere else already but the most promising post I can see is that it is fixed already....but no notes on how.

**Comment by weslambert** [comment]
Does the box have internet access?

**Reply by RagnarSTS** [reply]
Doesn't seem to have any issues at the moment. 
In a perfect world I would love to have a ports/protocols spreadsheet or set of rules I could put into a firewall/IDS system to ensure it only talks to exactly where it is supposed to...but it seems to be working normally and reaching out for updates and such.

**Reply by RagnarSTS** [reply]
/var/spool/cron and accompanying other items have that nice warning about being managed by Salt; and while I understand cron quite well Salt still somewhat mystifies me and I have no Salt command that I can trigger the geo update.

**Comment by RagnarSTS** [comment]
Still no proper working map data. Can anybody share the location and trigger mechanism for the script?  It would probably make troubleshooting this a lot faster as right now I'm assuming it triggers once a week...makes trial and error at fixing it rather slow.

**Comment by RagnarSTS** [comment]
Been a few months and there doesn't seem to be any movement on any of these sorts of index items.
I've seen a few posts such as [https://github.com/Security-Onion-Solutions/securityonion/discussions/2881](url) where somebody was able to fix their index patterns.  
Unfortunately the read-the-docs section  [https://docs.securityonion.net/en/2.3/re%E2%80%90indexing.html](url) doesn't actually instruct anybody how to re-index anything, and instead just provides an experimental script, which so far hasn't fixed any issues that I know of.
Default load of SO2 seems to be missing quite a few little items in the queries; such as "Could not locate that index-pattern-field (id:xxx)
dns.response.code_name.keyword
total_bytes
client.bytes
server.bytes
connection.history.keyword
ssh.authentication.success
smb.share_type.keyword
@timestamp

Unfortunately unlike the post above (#2881), I cannot gather the power to decipher the 4 bullet points and reindex note. I have no issues following pages and pages of painstaking instructions, I just can't find any that seem to match what I see in the the SO2 stack manager (7.11.2 right now).

**Reply by weslambert** [reply]
We don't describe the process of re-indexing in great detail because that starts to get into more generalized Elasticsearch management, for which there is plenty of information from the Elastic side.  As time allows, we can augment this, but many times there are certain priorities that take precedence.  To my knowledge, the main issue with the maps is that the geo fields are not correctly mapped/typed, so there is no way to render the map.  I have not had a chance to take a look back at this yet, due to competing priorities.  When you refer to queries, are you referring to the visualizations?

**Reply by RagnarSTS** [reply]
Thanks for the reply, and I definitely get it.  Yes the visualizations. 
I would suggest for the next release simply removing all the visualizations that don't work out of the box.  If one is performing a trial or test installation, having ~15% of the things not functioning isn't a selling point. While I get that the maps and some other things are NOT the most useful, sometimes functioning maps can be useful as a simple wow factor for decision makers. If the back end work doesn't function out of the box, then it no longer triggers the instant love feelings. Tuning, customizing, and making it super functional can take a few weeks; but one of the huge selling points really is quick deploy turnkey functionality out of the box. 
Even back on the release candidates I reinstalled it several times thinking I had somehow did something whack. Waited a week for a cron job to update geolocation data ....reinstalled.  etc.  Pretty sure I wasn't the only one.  If the built in visualization wasn't there, I'd probably just went about my business rather than thinking I'd somehow mucked it up. I only remembered the issue this week when I went to do a quick install/demo and obvious question came out on how to fix it. 
Cheers to all, and SO is still my favorite (mostly) turnkey solution. :)

**Comment by RagnarSTS** [comment, marked as answer]
Fixed in 2.3.110!
